### PR TITLE
submitting Allow unicode characters in download scripts

### PR DESCRIPTION
**Pull request by behrangi**
[#182782758]

### CODE REVIEW

**Comment by behrangi** [comment]
@awaisathar The task just mentioned download batch script, shall I consider zip files direct download as well? Or ui will consider it?

**Comment by awaisathar** [comment]
Thanks, I think we have to cover all download points including dynamically created zip downloads as well as folders.

File: src/main/java/uk/ac/ebi/biostudies/controller/FileDownload.java [hunk 1]
```diff
@@ -44,44 +52,55 @@ public class FileDownload {
...
         String[] files = request.getParameterMap().get("files");
-        if(dlType.equalsIgnoreCase("zip"))
+        if (dlType.equalsIgnoreCase("zip"))
             zipDownloadService.sendZip(request, response, files, storageMode);
-        else if(dlType.equalsIgnoreCase("ftp") || dlType.equalsIgnoreCase("aspera")){
-            response.setContentType("application/txt");
-            response.addHeader("Content-Disposition", "attachment; filename="+accession+"-" + os+"-"+dlType+"."+fileExtension);
+        else if (dlType.equalsIgnoreCase("ftp") || dlType.equalsIgnoreCase("aspera")) {
+            List<String> newFileNames = new ArrayList<>();
+            if (storageMode == Constants.File.StorageMode.FIRE) {
```
Comment (awaisathar) [inline src/main/java/uk/ac/ebi/biostudies/controller/FileDownload.java:74]: Extracting a function here would make this much more readable

File: src/main/java/uk/ac/ebi/biostudies/controller/FileDownload.java [hunk 2]
```diff
@@ -44,44 +52,55 @@ public class FileDownload {
...
+                    }
+                    return "";
+                }).collect(Collectors.toList());
+            }
+            response.setContentType("application/txt; charset=UTF-8");
+            response.addHeader("Content-Disposition", "attachment; filename=" + accession + "-" + os + "-" + dlType + "." + fileExtension);
             response.addHeader("Cache-Control", "no-cache");
-            response.getOutputStream().print(batchDownloadScriptBuilder.fillTemplate(dlType, fileNames, relativeBaseDir, os, storageMode));
-            response.getOutputStream().close();
+            response.getWriter().print(batchDownloadScriptBuilder.fillTemplate(dlType, storageMode == Constants.File.StorageMode.FIRE ? newFileNames : fileNames, relativeBaseDir, os, storageMode));
```
Comment (awaisathar) [inline src/main/java/uk/ac/ebi/biostudies/controller/FileDownload.java:87]: No check would be needed if we just use the newFileNames and do the replacement in a function (as specified above)

**Review by awaisathar — APPROVED** [review]
LGTM. Please see the minor updates and merge.

**Pull request merged**